### PR TITLE
Adding gcd 

### PR DESCRIPTION
**Pull request by MyatCharm**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [x] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations
- [x] The function is not called in the function file

### Do's

- [x] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] There are no commented lines of code
- [x] There are no `print` statements anywhere
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

**Comment by Akan186** [comment]
It appears that you failed 2 CI checks in your "gcd.py" Can you fix those?

File: solutions/tests/test_gcd.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import unittest
```
Comment (Akan186) [inline solutions/tests/test_gcd.py:8]: You need to add a class-level docstring above this.

File: solutions/gcd.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+"""Calculate the Greatest Common Divisor (GCD) of two numbers
```
Comment (Akan186) [inline solutions/gcd.py:1]: Include a type annotation

File: solutions/gcd.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        5
+        >>> gcd(18,63)
+        9
+
+    @author: Myat Charm
+    Created on Jan 03, 2025.
+    """
+    while b != 0:
+        a, b = b, a % b
+    return a
```
Comment (Akan186) [inline solutions/gcd.py:28]: Add a defensive assertion

**Review by Akan186 — CHANGES_REQUESTED** [review]

File: solutions/tests/test_gcd.py (hunk 1 above)
Comment (MyatCharm) [reply solutions/tests/test_gcd.py:8]: I'm on it

File: solutions/gcd.py (hunk 3 above)
Comment (MyatCharm) [reply solutions/gcd.py:28]: Thanks

File: solutions/gcd.py (hunk 2 above)
Comment (MyatCharm) [reply solutions/gcd.py:1]: fixing

**Review by Akan186 — APPROVED** [review]
I think that's pretty much it.
